### ISSUE TITLE
Unable to unzip file

### ISSUE THREAD

**Issue by anon0010** [HACKERALERT/Picocrypt#179]
I have a huge problem. I decrypted the volume and changed the extension to .zip like I always do, but somehow I can't open the file anymore. Can you please help me?

<img width="381" alt="image" src="https://github.com/HACKERALERT/Picocrypt/assets/167910573/bfa17f3f-c9a5-4d4a-a41c-b11a44b783e6">

**Comment by HACKERALERT** [comment]
There are two possibilities I can see:
1. The original file you encrypted was not a zip file or a folder. Are you sure that the original file was a zip file? Could it be a .pdf for example, but you removed it from the output filename and forgot?
2. Try extracting the zip file via 7-Zip or Keka/equivalent. This probably won't make a difference, but worth a try if you are absolutely certain that the original file was a .zip file.

If decrypting it in Picocrypt was successfull and gave no errors, I can tell you with pretty high certainty that it is not an encryption/decryption error because file integrity is checked and any corruption would've been caught by Picocrypt.

**Comment by anon0010** [comment]
I'm very sure it's a .zip file. I tried with different third-party archivers but none worked. I downloaded a backup that I store in the cloud rather than my USB, but it's the same issue. It's not a pdf file, but a whole folder that I encrypted using Picocrypt.

**Comment by HACKERALERT** [comment]
1. Did Picocrypt show any errors when decrypting
2. Did you use any advanced features/keyfiles
3. What version of Picocrypt did you use to encrypt?

**Comment by anon0010** [comment]
1. Did Picocrypt show any errors when decrypting
No, no errors.
3. Did you use any advanced features/keyfiles
No keyfiles, but probably Paranoid mode, compress files, reed-solomon, deniability.
5. What version of Picocrypt did you use to encrypt?
I think the latest. Last time that I modified the volume was around dec of 2023.

**Comment by anon0010** [comment]
It's saying "can't read header, assuming volume is deniable."

**Comment by anon0010** [comment]
Then it's decrypting successfully, what I always do is change the extension to .zip and all the time it just unzips it, but now it won't.

**Comment by HACKERALERT** [comment]
Can you try a random test folder, encrypt it with the same parameters as you did with the broken .zip and decrypt it again to see if it reproduces your current issue?

**Comment by anon0010** [comment]
Just tried and no issues and I can't reproduce it. I'm 100% sure it's a .zip file. I used it all the time. There can not be something else :/

**Comment by HACKERALERT** [comment]
I'm sorry but I can't think of anything else to try then. I'll leave this issue open and if anyone has similiar experiences to share or help me debug with, I will be happy to listen.

**Comment by anon0010** [comment]
I have a huge problem if I can't open this. Please keep me posted if something comes up in your mind. Thanks!

**Comment by BigPanda97** [comment]
Already tried tools like "DiskInternals ZIP Repair" or "GetData ZIP Repair"?

Maybe the zip was damaged before and can be repaired this way.

**Comment by anon0010** [comment]
They were compressed using Picocrypt on macOS. Those applications are for Windows AFAIK. Thanks for your reply though. I really appreciate all the help!

**Comment by HACKERALERT** [comment]
Also worth a try:
https://gildas-lormeau.github.io/zip.js/demos/demo-read-file.html

**Comment by anon0010** [comment]
Thanks for your help. 

It's throwing this error: `Error: End of central directory not found`

As for the hex editor, this is what i see:

<img width="408" alt="image" src="https://github.com/HACKERALERT/Picocrypt/assets/167910573/1c520a62-2728-483c-87aa-24b0fd31b63a">

**Comment by anon0010** [comment]
The size of the file is correct. By the way. Not sure if that helps.

**Comment by HACKERALERT** [comment]
@anon0010 Do you have any older copies of your volume? If they decrypt properly, is there something new that you did such as using a different set of advanced features, encrypting/decrypting from different storage locations that could've affected your latest copy?

**Comment by anon0010** [comment]
The strange thing is that I always unzipped it and zipped it using Picocrypt. How can I resolve this? This is an extremely important file. I created two backups, one in the cloud and one on my usb drive, but both doesn't seem to work.

@HACKERALERT no, i made a backup as redundancy with the hope that if I ever lose the file I can get it from cloud storage.. but they're not working. i tried to fire up a VM with a clean macOS to see if that made a difference as well, i compiled it with your instructions to Silicon chipset instead of using Rosetta (although any changes would be extremely minimal), but nothing works so far and it's really breaking my head.

**Comment by anon0010** [comment]
@CodeCracker-oss if you know that `50 4b 03 04` is a zip file, could you by any chance know what might be the first 4 bytes of my file? i mean i'm 100% sure it's a zip file, but maybe we can debug it a bit more?

**Comment by anon0010** [comment]
I got it.

**Comment by anon0010** [comment]
I found out that months ago I used zip file to encrypt certain data and I changed it too KeePass database instead, but totally forgot about that.  With your info about the 4 bytes, I checked the 4 bytes of my files in Google and found out that it could be Cryptomator or a KeePass database. This reminded me that I changed it to a KeePass database back then. I really thought it was a zip file, but I totally forgot that I changed it. Feels stupid lol! Nevertheless, I want to thank you guys for solving this for me and for all your support! Thanks!
